### PR TITLE
Switch to CircleCI 2.0

### PR DESCRIPTION
**Pull request by enykeev**
So we can start using st2-docker as our test environment.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+# IS_ENTERPRISE
+# PACKAGECLOUD_ORGANIZATION (default: stackstorm)
+# PACKAGECLOUD_TOKEN
+
+version: 2
+jobs:
+  build:
+    machine: true
+    node:
+      version: 4.2.2
```
Comment (armab) [inline .circleci/config.yml]: I'm afraid, this doesn't work in Circle 2.0 anymore (machine executor)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      ST2_USERNAME: admin
+      ST2_PASSWORD: 123
+    steps:
+      - checkout
+      - run:
+          name: Update Docker Compose
+          command: |
+            set -x
+            sudo sh -c "curl -L https://github.com/docker/compose/releases/download/1.14.0/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose"
+            sudo chmod +x /usr/local/bin/docker-compose
```
Comment (armab) [inline .circleci/config.yml]: Docker-compose is already installed, we can remove this step

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      - run:
+          name: Create directories for Artifacts
+          command: mkdir ~/artifacts && cd ~/artifacts && mkdir ${DEB} ${RPM} logs
+      - run:
+          name: Download helper scripts (for packagecloud)
+          command: |
+            mkdir ~/scripts
+            wget -qO - https://github.com/StackStorm/st2-packages/raw/master/.circle/packagecloud.sh > ~/scripts/packagecloud.sh
+            wget -qO - https://github.com/StackStorm/st2-packages/raw/master/.circle/save_payload.py > ~/scripts/save_payload.py
```
Comment (armab) [inline .circleci/config.yml]: Can delete this `save_payload`. Not used anywhere.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            echo $RPM | tr ' ' '\n' | xargs -I{} cp -vr ../st2web-*.rpm ~/artifacts/{}
+      - store_artifacts:
+          path: ~/artifacts
+          destination: packages
+      - deploy:
+          name: Maybe Deploy
+          command: |
+            if [[ "${CIRCLE_BRANCH}" == "master" || "${CIRCLE_BRANCH}" =~ v[0-9]+(\.[0-9]+)* || "${CIRCLE_BRANCH}" == "feature/circleci" ]]; then
+              for distro in ${RPM} ${DEB}; do
+                packagecloud.sh deploy $distro $CIRCLE_ARTIFACTS/$distro
```
Comment (armab) [inline .circleci/config.yml]: Can you double check `$CIRCLE_ARTIFACTS` env? I'm afraid that variable is not available anymore in 2.0.

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+          command: |
+            PKG_VERSION=$(node -e "console.log(require('./package.json').st2_version);")
+            PKG_RELEASE=$(packagecloud.sh next-revision trusty ${PKG_VERSION} st2web)
+            echo "export PKG_VERSION=${PKG_VERSION}" >> $BASH_ENV
+            echo "export PKG_RELEASE=${PKG_RELEASE}" >> $BASH_ENV
+      - run:
+          name: Install system dependencies
+          command: |
+            sudo apt-get update
+            sudo apt-get install rpm parallel jq devscripts debhelper
```
Comment (armab) [inline .circleci/config.yml]: `parallel` could be removed as not used anywhere

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+            wget -qO - https://github.com/StackStorm/st2-packages/raw/master/.circle/save_payload.py > ~/scripts/save_payload.py
+            chmod 755 ~/scripts/{packagecloud.sh,save_payload.py}
+            echo "export PATH=${HOME}/scripts:${PATH}" >> $BASH_ENV
+      - run:
+          name: Export package version
+          command: |
+            PKG_VERSION=$(node -e "console.log(require('./package.json').st2_version);")
+            PKG_RELEASE=$(packagecloud.sh next-revision trusty ${PKG_VERSION} st2web)
+            echo "export PKG_VERSION=${PKG_VERSION}" >> $BASH_ENV
+            echo "export PKG_RELEASE=${PKG_RELEASE}" >> $BASH_ENV
```
Comment (armab) [inline .circleci/config.yml]: :+1:

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+          name: Update env variables for test containers
+          command: |
+            make -C ~/st2-docker env
+            echo -e "ST2_USER=${ST2_USERNAME}\nST2_PASSWORD=${ST2_PASSWORD}" > ~/st2-docker/conf/stackstorm.env
+            cat ~/st2-docker/conf/stackstorm.env
+      - run:
+          name: Start test containers
+          command: |
+            docker-compose -f ~/st2-docker/docker-compose.yml up -d
+            sleep 60
```
Comment (armab) [inline .circleci/config.yml]: So yeah, as you said this is nasty indeed.

@warrenvw maybe you can suggest some solution for this?

File: .circleci/config.yml (hunk 1 above)
Comment (enykeev) [inline .circleci/config.yml]: Ack. Will check.

File: .circleci/config.yml (hunk 2 above)
Comment (enykeev) [inline .circleci/config.yml]: https://circleci.com/gh/StackStorm/st2web/354#config/containers/0

```
#!/bin/bash -eo pipefail
docker -v
docker-compose -v

Docker version 17.03.0-ce, build 60ccb22
docker-compose version 1.9.0, build 2585387
```

File: .circleci/config.yml (hunk 4 above)
Comment (enykeev) [inline .circleci/config.yml]: yep, good catch.

File: .circleci/config.yml (hunk 2 above)
Comment (armab) [inline .circleci/config.yml]: Makes sense, thanks for checking!

File: .circleci/config.yml (hunk 1 above)
Comment (enykeev) [inline .circleci/config.yml]: Indeed, doesn't work anymore.


https://circleci.com/gh/StackStorm/st2web/378
```
npm WARN engine zombie@4.1.0: wanted: {"node":"^1.6.0"} (current: {"node":"6.1.0","npm":"2.15.12"})
```

**Pull request merged**